flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/RowSerializer.java

```diff
@@ -71,19 +82,34 @@
 
 	private final int arity;
 
+	private final @Nullable LinkedHashMap<String, Integer> positionByName;
+
 	private transient boolean[] mask;
 
+	private transient Row reuseRowPositionBased;
+
 	public RowSerializer(TypeSerializer<?>[] fieldSerializers) {
-		this(fieldSerializers, false);
+		this(fieldSerializers, null, false);
+	}
+
+	public RowSerializer(
+			TypeSerializer<?>[] fieldSerializers,
+			@Nullable LinkedHashMap<String, Integer> positionByName) {
+		this(fieldSerializers, positionByName, false);
 	}
 
 	@SuppressWarnings("unchecked")
-	public RowSerializer(TypeSerializer<?>[] fieldSerializers, boolean legacyModeEnabled) {
+	public RowSerializer(
+			TypeSerializer<?>[] fieldSerializers,
+			@Nullable LinkedHashMap<String, Integer> positionByName,
+			boolean legacyModeEnabled) {
```

Comment:
I was wondering if it wouldn't make sense to drop legacy mode now, for Flink 1.13.

Reply:
+1 I can open a PR for it